spring-integration-mqtt/src/main/java/org/springframework/integration/mqtt/inbound/Mqttv5PahoMessageDrivenChannelAdapter.java

```diff
@@ -106,6 +109,45 @@ public Mqttv5PahoMessageDrivenChannelAdapter(MqttConnectionOptions connectionOpt
 		}
 	}
 
+	/**
+	 * Use this constructor when you need to use a single {@link ClientManager}
+	 * (for instance, to reuse an MQTT connection).
+	 * @param clientManager The client manager.
+	 * @param topic The topic(s).
+	 */
+	public Mqttv5PahoMessageDrivenChannelAdapter(ClientManager<IMqttAsyncClient> clientManager, String... topic) {
+		this(buildDefaultConnectionOptions(null), clientManager, topic);
+	}
+
+	/**
+	 * Use this constructor when you need to use a single {@link ClientManager}
+	 * (for instance, to reuse an MQTT connection) and a specific {@link MqttConnectionOptions}.
+	 * @param connectionOptions The connection options.
+	 * @param clientManager The client manager.
+	 * @param topic The topic(s).
+	 */
+	public Mqttv5PahoMessageDrivenChannelAdapter(MqttConnectionOptions connectionOptions,
+			ClientManager<IMqttAsyncClient> clientManager, String... topic) {
+
+		super(clientManager, topic);
+		this.connectionOptions = connectionOptions;
+		if (!this.connectionOptions.isAutomaticReconnect()) {
```

Comment:
This is not.
In case of an external `ClientManager`, the reconnection logic is handled over there.
So, our particular channel  adapter must not do that  any more.

Reply:
Not exactly. This one is required to successfully subscribe to the topics *initially*. If we cannot do it initially - no sense in automatic reconnect when we don't have anything to listen to 🙂 
This piece was in place initially, you can check current code as well, `doStart` method.

Reply:
Correct it was here originally since we create the client internally.
Now we provide that client manager which has a logic to create, connect and reconnect the client.
If the client manager is provided, we must not handle this logic over here: we have to rely on whatever is provided by the client manager.

We may show WARN in logs that reconnection must be set in the client since it is convenient for this type of channel adapters, but we must not do that over here.
By the way, the `phase` for auto-start on that client manager must be less than `Integer.MAX_VALUE / 2` to let it to be started exactly *before* this message-driven channel adapter.
Better to have it `0` by default and let end-user to set their own desired. With respective JavaDoc, of course, to mention that phase of this has to be less than phase of message-driven channel adapter.
That's probably your concern since in tests you see that `Mqttv5PahoMessageDrivenChannelAdapter` is started before provided `ClientManager`. Just because the default is:
```

	/**
	 * The default phase for {@code SmartLifecycle}: {@code Integer.MAX_VALUE}.
	 * <p>This is different from the common phase {@code 0} associated with regular
	 * {@link Lifecycle} implementations, putting the typically auto-started
	 * {@code SmartLifecycle} beans into a later startup phase and an earlier
	 * shutdown phase.
	 * @since 5.1
	 * @see #getPhase()
	 * @see org.springframework.context.support.DefaultLifecycleProcessor#getPhase(Lifecycle)
	 */
	int DEFAULT_PHASE = Integer.MAX_VALUE;
```

Reply:
Nice, thanks for explanation, I will take a look at it!

Reply:
Seems like tests pass without the need in additional manual `reconnect` call. Resolving this then.